### PR TITLE
Feature EPL-565: add default filters in DAS microservice over native query

### PR DESCRIPTION
**Pull request by giuliodoriEtendo**

### CODE REVIEW

File: modules_core/com.etendorx.das/src/main/java/com/etendorx/das/hibernate_interceptor/CustomInterceptor.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import static com.etendorx.utils.auth.key.context.AppContext.getCurrentUser;
+
+
+@Component
+@Slf4j
+public class CustomInterceptor extends EmptyInterceptor {
+
+    @Override
+    public String onPrepareStatement (String sql) {
+        String adUser = getCurrentUser().getUserId();
```
Comment (Gremiger) [inline modules_core/com.etendorx.das/src/main/java/com/etendorx/das/hibernate_interceptor/CustomInterceptor.java:17]: Podría llamarse userId?

File: modules_core/com.etendorx.das/src/main/java/com/etendorx/das/utils/DefaultFilters.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        } else {
+            resultSql = sql.replace("limit", "where "+ tableName +".ad_client_id = '"+ clientId +"' " +
+                    "and "+ tableName +".ad_org_id = ANY (etrx_org_tree('"+ orgId +"')) " +
+                    "and "+ tableName +".isactive = '"+ isActive +"' limit");
+        }
+        return resultSql;
+    }
+
+    private static boolean isSuperUser(String adUser, String clientId, String orgId) {
+        if (adUser.compareTo("100") == 0 && clientId.compareTo("0") == 0 && orgId.compareTo("0") == 0 ) {
```
Comment (Gremiger) [inline modules_core/com.etendorx.das/src/main/java/com/etendorx/das/utils/DefaultFilters.java:43]: Usar StringUtils.equals o YodaCondition.
 Es preferible usar equals directamente en lugar del compareTo si no me equivoco.

"100".equals(adUser)

o

StringUtils.equals(adUser, "100")

File: modules_core/com.etendorx.das/src/main/java/com/etendorx/das/utils/DefaultFilters.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+@Slf4j
+public class DefaultFilters {
+    public static String addFilters(String sql, String adUser, String clientId, String orgId, String isActive) {
+
+        //SUPERUSER BY PASS FILTERS
+        if (isSuperUser(adUser, clientId, orgId)) return sql;
```
Comment (juabaez) [inline modules_core/com.etendorx.das/src/main/java/com/etendorx/das/utils/DefaultFilters.java:15]: always use Curly brace

File: modules_core/com.etendorx.das/src/main/java/com/etendorx/das/utils/DefaultFilters.java [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+        boolean containsWhere = sql.contains("where");
```
Comment (juabaez) [inline modules_core/com.etendorx.das/src/main/java/com/etendorx/das/utils/DefaultFilters.java:17]: move "where" string to a constant

File: modules_core/com.etendorx.das/src/main/java/com/etendorx/das/utils/DefaultFilters.java (hunk 2 above)
Comment (juabaez) [inline modules_core/com.etendorx.das/src/main/java/com/etendorx/das/utils/DefaultFilters.java:43]: user StringUtils to compare

**Review by juabaez — CHANGES_REQUESTED** [review]

**Comment by sebastianbarrozo** [comment]
Please, squash all commits into one in order to have a better description

File: modules_core/com.etendorx.das/src/main/java/com/etendorx/das/utils/DefaultFilters.java [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+    private static boolean isSuperUser(String adUser, String clientId, String orgId) {
+        if (adUser.compareTo("100") == 0 && clientId.compareTo("0") == 0 && orgId.compareTo("0") == 0 ) {
+            return true;
+        }
+        return false;
+    }
+
+    private static String getTableName(String sql) {
+        Pattern selectPattern = Pattern.compile("select\\s+(\\w+)\\.");
```
Comment (Matias-Bernal) [inline modules_core/com.etendorx.das/src/main/java/com/etendorx/das/utils/DefaultFilters.java:50]: private static String getTableName(String sql) { 
        Pattern pattern = Pattern.compile("select\\s+(\\w+)\\.");
        Matcher matcher = pattern.matcher(sql);
        if (matcher.find()) {
            return matcher.group(1);
        } else {
            pattern = Pattern.compile("select\\s+count\\(\\s*(\\w+)\\.\\w+\\)\\s+as");
            matcher = pattern.matcher(sql);
            if (matcher.find()) {
                return matcher.group(1);
            }
        }
        log.error("[ getTableName ] - PATTERN ERROR");
        throw new QueryException("getTableName ERROR ");
  }

 Less vars and less patters's checks.

**Review by Matias-Bernal — DISMISSED** [review]

File: modules_core/com.etendorx.das/src/main/java/com/etendorx/das/utils/DefaultFilters.java (hunk 3 above)
Comment (giuliodoriEtendo) [reply modules_core/com.etendorx.das/src/main/java/com/etendorx/das/utils/DefaultFilters.java]: done

File: modules_core/com.etendorx.das/src/main/java/com/etendorx/das/utils/DefaultFilters.java (hunk 2 above)
Comment (giuliodoriEtendo) [reply modules_core/com.etendorx.das/src/main/java/com/etendorx/das/utils/DefaultFilters.java]: done

File: modules_core/com.etendorx.das/src/main/java/com/etendorx/das/hibernate_interceptor/CustomInterceptor.java (hunk 1 above)
Comment (giuliodoriEtendo) [reply modules_core/com.etendorx.das/src/main/java/com/etendorx/das/hibernate_interceptor/CustomInterceptor.java]: done

File: modules_core/com.etendorx.das/src/main/java/com/etendorx/das/utils/DefaultFilters.java (hunk 5 above)
Comment (giuliodoriEtendo) [reply modules_core/com.etendorx.das/src/main/java/com/etendorx/das/utils/DefaultFilters.java]: getTableName method was refactorized to getTableAlias, and uses only 1 matching pattern

File: modules_core/com.etendorx.das/src/main/java/com/etendorx/das/hibernate_interceptor/CustomInterceptor.java [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+@Slf4j
+public class CustomInterceptor extends EmptyInterceptor {
+
+    @Override
+    public String onPrepareStatement (String sql) {
+        String userId = getCurrentUser().getUserId();
+        String clientId = getCurrentUser().getClientId();
+        String orgId = getCurrentUser().getOrganizationId();
+        String isActive = "Y";
+        String finalQuery = DefaultFilters.addFilters(sql, userId, clientId, orgId, isActive);
```
Comment (juabaez) [inline modules_core/com.etendorx.das/src/main/java/com/etendorx/das/hibernate_interceptor/CustomInterceptor.java:21]: the code style is fine ? can you check that ?

File: modules_core/com.etendorx.das/src/main/java/com/etendorx/das/utils/DefaultFilters.java [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+            //ADD DEFAULT FILTERS
+            String resultSql = replaceInQuery(sql, clientId, orgId, isActive, containsWhere, tableAlias);
+            return resultSql;
+    }
+
+    @NotNull
+    private static String replaceInQuery(String sql, String clientId, String orgId, String isActive, boolean containsWhere, String tableAlias) {
+        String resultSql;
+        if (containsWhere) {
+            resultSql = sql.replace(" where ", " where "+ tableAlias +".ad_client_id = '"+ clientId +"' " +
```
Comment (juabaez) [inline modules_core/com.etendorx.das/src/main/java/com/etendorx/das/utils/DefaultFilters.java:39]: move "where" as constant

**Review by juabaez — CHANGES_REQUESTED** [review]

File: modules_core/com.etendorx.das/src/main/java/com/etendorx/das/hibernate_interceptor/CustomInterceptor.java (hunk 6 above)
Comment (giuliodoriEtendo) [reply modules_core/com.etendorx.das/src/main/java/com/etendorx/das/hibernate_interceptor/CustomInterceptor.java:21]: yes

File: modules_core/com.etendorx.das/src/main/java/com/etendorx/das/utils/DefaultFilters.java (hunk 7 above)
Comment (giuliodoriEtendo) [reply modules_core/com.etendorx.das/src/main/java/com/etendorx/das/utils/DefaultFilters.java]: done

**Review by Matias-Bernal — APPROVED** [review]

**Comment by valeriag-etendo** [comment]
@giuliodoriEtendo The format of the PullRequest name must be:
`Feature EPL-565: <description>`

**Review by Gremiger — APPROVED** [review]

**Comment by valeriag-etendo** [comment]
@giuliodoriEtendo Don't forget to check the Sonar comments and correct the formatting of the commits.

**Review by valeriag-etendo — APPROVED** [review]

**Pull request merged**